rasa/core/run.py

```diff
@@ -180,8 +187,12 @@ def serve_application(
     from rasa.core import broker
 
     try:
-        _, nlu_model = get_model_subdirectories(get_model(model_path))
+        unpacked_model = get_model(model_path)
+        _, nlu_model = get_model_subdirectories(unpacked_model)
         _interpreter = NaturalLanguageInterpreter.create(nlu_model, endpoints.nlu)
+
+        shutil.rmtree(unpacked_model)
```

Comment:
Could we write a context manager for this: https://docs.python.org/3/library/contextlib.html?

E.g. that we would do 

```
with get_model(model_path) as unpacked_model:
   unpacked_model = get_model(model_path)
    _, nlu_model = get_model_subdirectories(unpacked_model)
```

Reply:
That's a good idea - should I change the behaviour of `get_model` and adapt all it's uses, or create a new function? I think the former is riskier since I'll need to much more code (I count 16 uses of the function).

Reply:
We could also do both at the same time, and modify `get_model` so that it returns an instance of:
```python
class UnpackedModelPath(str):
    def __enter__(self):
        return self

    def __exit__(self, _exc, _value, _tb):
        shutil.rmtree(self)
```

Which would delete the path it points to only when used as a context manager (but otherwise it would be a normal string).

Reply:
that is brilliant 🥇 Yes please! and please a test for the context manager use 👍